pyproject.toml

```diff
@@ -1,3 +1,39 @@
+[project]
+name = "mimir-worker-k8s"
+version = "0.0"  # this is in fact irrelevant
+requires-python = ">=3.12"
+
+dependencies = [
+    "ops>=1.5.0",
+    # Charm relation interfaces
+    "pydantic",
+    
+    # ---PYDEPS---
+    "cosl>=0.0.43",
+    # lib/charms/tempo_coordinator_k8s/v0/charm_tracing.py
+    "opentelemetry-exporter-otlp-proto-http==1.21.0",
+]
+
+[project.optional-dependencies]
+lint = [
+    "ruff",
+    "codespell",
+]
+fmt = ["ruff"]
+static = ["pyright"]
```

Comment:
I'm in favor of collapsing these into a single group, similar to what @PietroPasotti had

Reply:
could group:
1. lint, static, fmt -> lint
2. unit, scenario -> unit
3. integration -> 
or into 1 called `dev`

Reply:
Run some unit tests and see if pytest-operator creates a model first, before running linting tests.